frontend/src/metabase/components/NewItemMenu/NewItemMenu.tsx

```diff
@@ -130,6 +131,15 @@ const NewItemMenu = ({
       });
     }
 
+    if (hasDataAccess) {
```

Comment:
Even though #37352 says
> Add new option to the menu. **It should match visibility settings of models & questions.**

I've used only data access as a prerequisite here. Models have an additional requirement for the native write permissions, but that's because you can choose to create a model either from a GUI or from a native query. 

Metrics v2 are not possible to create from the native queries. At least not for the forseeable future.